src/Integration/Suppression/SonarQubeIssuesProvider.cs

```diff
@@ -156,11 +171,12 @@ private async Task SynchronizeSuppressedIssuesAsync()
                 var moduleKeyToRelativePathToRoot = (await this.sonarQubeService.GetAllModulesAsync(sonarQubeProjectKey,
                         cancellationTokenSource.Token))
                     .ToDictionary(x => x.Key, x => x.RelativePathToRoot ?? string.Empty);
+                this.hasModules = moduleKeyToRelativePathToRoot.Keys.Count == 1;
 
                 this.cachedSuppressedIssues = (await this.sonarQubeService.GetSuppressedIssuesAsync(sonarQubeProjectKey,
                         cancellationTokenSource.Token))
-                    .Select(x => ToIssueWithFixedRelativePath(moduleKeyToRelativePathToRoot, x))
-                    .GroupBy(x => x.FilePath)
+                    .Select(x => new { Key = ProcessKey(moduleKeyToRelativePathToRoot, x), Issue = x })
```

Comment:
If I understand correctly, the keys of items in the cache will be a mix of file-based keys and non-file based keys. When we're doing a lookup into the list we're walking the keys doing two comparisons per key, one to see if it's a file-based key and a second if it isn't.
It would be more efficient if we maintained separate lists of file-based and non-file based keys and then when looking for a match we only seached one list - we'd be searching through fewer items, and only doing one comparison per item.
Even if this change is worth making, I think it should be done in a separate PR i.e. stabilise and test the work in progress, then worry about optimisation.